### ISSUE TITLE
look into getting predictions in parallel

### ISSUE THREAD

**Issue by ClimbsRocks** [ClimbsRocks/auto_ml#124]
If we're handed a list of 1,000,000 items, and asked to get predictions on them, we've already parallelized the process of getting predictions from the subpredictors, but we haven't applied the same logic to our main predictor. 

put this logic inside FinalModelATC. 

check to make sure our predictions list is longer than 100, to make sure this overhead is reasonable. 

if so, split our dataset up into ncpu chunks. or split it into chunks of 100. 

either way, make a list of these sub-chunks of our dataset. 
run map on that list, then make sure to get this whole list as a single layer at the end (1-d list, not nested lists).

**Comment by ClimbsRocks** [comment]
leaving this for the user to figure out.
